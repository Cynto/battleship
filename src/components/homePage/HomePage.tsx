import React from 'react';
import './homepage.css';
import { Link } from 'react-router-dom';

function HomePage(props: any) {
  return (
    <div data-testid="start-page" className="start-page">
      <div className="menu-container">
        <p>Your mission is to destroy all enemy ships!</p>
        <Link to="/start">
          <button>Start Game</button>
        </Link>
      </div>
    </div>
  );
}

export default HomePage;
