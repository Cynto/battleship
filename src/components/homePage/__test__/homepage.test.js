import React from 'react';
import ReactDOM from 'react-dom';
import HomePage from '../HomePage';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';

describe('Homepage tests', () => {
  afterEach(cleanup);
  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <HomePage />
      </Router>,
      div,
    );
  });
  test('Renders main container correctly', () => {
    const { getByTestId } = render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(getByTestId('start-page')).toHaveClass('start-page');
  });
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <HomePage />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
