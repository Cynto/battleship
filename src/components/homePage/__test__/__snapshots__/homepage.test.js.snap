// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage tests Matches snapshot 1`] = `
<div
  className="start-page"
  data-testid="start-page"
>
  <div
    className="menu-container"
  >
    <p>
      Your mission is to destroy all enemy ships!
    </p>
    <a
      href="#/start"
      onClick={[Function]}
    >
      <button>
        Start Game
      </button>
    </a>
  </div>
</div>
`;
